tests/tasks/setup/eks/awscli-pod-identity-association.yaml

```diff
@@ -0,0 +1,87 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: awscli-eks-pia-create
+  namespace: scalability
+spec:
+  description: |
+    Create an EKS Pod Identity Association for a given cluster.
+    This Task can be used to create an EKS Pod Identity Association for namespace default and service account default.
+  params:
+  - name: cluster-name
+    description: The name of the EKS cluster you want to create an EKS Pod Identity Association for.
+  - name: region
+    default: "us-west-2"
+    description: The region where the cluster is in.
+  - name: endpoint
+    default: ""
+  - name: namespace-prefix
+    default: "default"
+  - name: namespaces
+    default: "1"
+  workspaces:
+  - name: config
+    mountPath: /config/
+  stepTemplate:
+    env:
+    - name: KUBECONFIG
+      value: /config/kubeconfig
+  steps:
+  - name: write-kubeconfig
+    image: alpine/k8s:1.31.5
+    script: |
+      ENDPOINT_FLAG=""
+      if [ -n "$(params.endpoint)" ]; then
+        ENDPOINT_FLAG="--endpoint $(params.endpoint)"
+      fi
+      aws eks $ENDPOINT_FLAG update-kubeconfig --name $(params.cluster-name) --region $(params.region)
+  - name: create-pia
+    image: alpine/k8s:1.31.5
+    script: |
+      ENDPOINT_FLAG=""
+      if [ -n "$(params.endpoint)" ]; then
+        ENDPOINT_FLAG="--endpoint $(params.endpoint)"
+      fi
+
+      MANAGED_POLICY_ARN="arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess"
+      TRUST_POLICY_FILE="pia-trust-policy.json"
+      # create a trust policy json file
+      cat > $TRUST_POLICY_FILE <<EOF
```

Comment:
let's put trust policy  under assets here https://github.com/awslabs/kubernetes-iteration-toolkit/tree/main/tests/assets  and provide a url to pull it from there ?

we want to decouple this from the task def.

Reply:
done